src/main/java/com/google/step/snippet/servlets/UserSettingServlet.java

```diff
@@ -24,11 +32,67 @@
 /** Servelet that renders user setting page */
 @WebServlet("/user")
 public class UserSettingServlet extends HttpServlet {
+  private static final String WEBSITE_PARAMETER = "website";
+  private static final String LANGUAGE_PARAMETER = "language";
+  private static final String ID_PARAMETER = "id";
+  private static final String EMAIL_PARAMETER = "email";
+  private static final String USER_CLASS_PARAMETER = "UserInfo";
+  private static final String USER_PARAMETER = "user";
+  private String referer = "";
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response)
       throws IOException, ServletException {
+    UserService userService = UserServiceFactory.getUserService();
     // Forward the request to the template (which is a servlet itself).
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    String id = userService.getCurrentUser().getUserId();
+    String email = userService.getCurrentUser().getEmail();
+    Query.FilterPredicate filterUser =
+        new Query.FilterPredicate(ID_PARAMETER, FilterOperator.EQUAL, id);
+    Query queryUser = new Query(USER_CLASS_PARAMETER).setFilter(filterUser);
+    Entity userEntity = datastore.prepare(queryUser).asSingleEntity();
+    String website;
+    String language;
+    if (userEntity == null) {
+      website = null;
+      language = null;
+    } else {
+      website = (String) userEntity.getProperty(WEBSITE_PARAMETER);
+      language = (String) userEntity.getProperty(LANGUAGE_PARAMETER);
+    }
+
+    User u = new User(id, email, website, language);
+    request.setAttribute(USER_PARAMETER, u);
     request.getRequestDispatcher("WEB-INF/templates/user_dashboard.jsp").forward(request, response);
+    referer = request.getHeader("referer");
```

Comment:
So just always return sendRedirect("/user") here now, if I understood correctly.

And given this, do we need another button to go back to the snippet home page or srp now?